software/wmla120_ansible/validate_firewall_setup.yml

```diff
@@ -2,61 +2,68 @@
 #If a firewall is enabled, the following default ports must be granted access on all management hosts for IBM Spectrum Conductor Deep Learning Impact: 9243, 9280, 5000, 5001, 27017, and 6379. If you change these ports after installation, make sure to update firewall rules accordingly.
 #
 
-- name: Setting Variables
-  set_fact:
-    port_9243: '9243/tcp'
-    port_9080: '9080/tcp'
-    port_5000: '5000/tcp'
-    port_5001: '5001/tcp'
-    port_27017: '27017/tcp'
-    port_6379: '6379/tcp'
+- name: Include vars of envs_spectrum_conductor_dli.yml into the 'ports' variable
+  include_vars:
+    file: envs_spectrum_conductor_dli.yml
+    name: ports
 
-- name: List all host firewall data
-  shell: "firewall-cmd --list-all"
+- name: DEBUG - List imported variables
+  debug:
+    msg: "{{ ports }}"
+
+- name: Check status of firewall
+  shell: "firewall-cmd --state"
   delegate_to: localhost
   register: host_firewall_status
-  become: yes
+  become: yes  
+  ignore_errors: yes
 
-- name: Checking firewall_status stdout
+- name: DEBUG - Checking firewall_data stdout
   debug:
     msg: "{{ host_firewall_status.stdout }}"
 
-- name: Checking firewall_status for http service enabled
+- name: List all host firewall data
+  shell: "firewall-cmd --list-services"
+  delegate_to: localhost
+  when: host_firewall_status.stderr != 'not running'
+  register: host_firewall_data
+  become: yes
+
+- name: DEBUG - Checking firewall_data stdout
+  debug:
+    msg: "{{ host_firewall_data.stdout }}"
+  when: host_firewall_status.stderr != 'not running'
+
+- name: DEBUG - Checking firewall_data for http service enabled
   debug:
-    msg: "{{ host_firewall_status.stdout.find('http') }}"
+    msg: "{{ host_firewall_data.stdout.find('http') }}"
+  when: host_firewall_status.stderr != 'not running'
 
-- name: Print success status of Host firewall
-  command: echo "Firewall Exists"
+- name: Print success status of host firewall
+  command: echo "Firewall Exists and http enabled"
   when: host_firewall_status.stdout.find('http') != -1
 
-- name: Else print fail status
-  command: echo "Firewall Does Not Exist"
-  when: host_firewall_status.stdout.find('http') == -1
+- name: Else print fail status 
+  command: echo "WARNING - Firewall Not Enabled"
+  when: host_firewall_status.stderr == 'not running'
 
-- name: Open Ports for Spectrum Conductor
+- name: Open Ports for Spectrum Conductor - If firewall is enabled
```

Comment:
Overall the playbook is looking pretty good.  I hadn't previously noticed that you were using the ansible firewalld module.  This is actually enabling the ports even if they're not already. Since this is a validate playbook it shouldn't modify the existing setup. So please run firewall-cmd --list-all or simply firewall-cmd --list-ports and check the results.  BTW Save this task (maybe just comment it out) so we can use it in a config playbook

Reply:
thanks!
okay, i have added the following task to list back port information:
 
- name: Check Enabled ports status 
  command: "firewall-cmd --list-ports"
  when: host_firewall_status.stderr != 'not running'
  register: port_status
  
- name: DEBUG - Check Enabled ports status 
  debug:
    msg: "{{ port_status }}"